TOOLS/macos-sdk-version.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+# Logic copied from compiler_swift.py in the waf build. This checks for the sdk
+# path, the sdk version, and the sdk build version. The sdk path is returned
+# along with what is selected as the sdk version.
+
+import re
+import os
+import string
+import sys
+from distutils.version import StrictVersion
+from shutil import which
+from subprocess import check_output
+
+def find_macos_sdk():
+    sdk = os.environ.get('MACOS_SDK', '')
+    sdk_version = os.environ.get('MACOS_SDK_VERSION', '0.0')
+    xcrun = which('xcrun')
+
+    if not xcrun:
+        return sdk,sdk_version
+
+    if not sdk:
+        sdk = check_output([xcrun, '--sdk', 'macosx', '--show-sdk-path'],
+                            encoding="UTF-8")
+    
+    # find macOS SDK paths and version
+    if sdk_version == '0.0':
+        # show-sdk-build-version: is not available on older command line tools, but returns a build version (eg 17A360)
+        # show-sdk-version: is always available, but on older dev tools it's only the major version
+        sdk_build_version = check_output([xcrun, '--sdk', 'macosx',
+                                          '--show-sdk-build-version'], encoding="UTF-8")
+
+        sdk_version = check_output([xcrun, '--sdk', 'macosx', '--show-sdk-version'],
+                                    encoding="UTF-8")
+
+    if sdk:
+        build_version = '10.10.0'
+
+    if sdk_version == '0.0':
```

Comment:
this check seems wrong. "sdk_version == '0.0'" can never be true because the variable is filled in line 34, effectively making the whole version check inside unreachable.

in the original code it didn't check for the output if xcrun but what was passed as an evn var https://github.com/mpv-player/mpv/blob/6265724f3331e3dee8d9ec2b6639def5004a5fa2/waftools/detections/compiler_swift.py#L148

Reply:
Ah yeah, this check is wrong and not needed. There's no global environment to store anything. It just returns.